### ISSUE TITLE
Unit tests fail on alembic 1.7

### ISSUE THREAD

**Issue by soininen** [Spine-project/Spine-Database-API#140]
`TestMigration.test_upgrade_content` will fail with alembic 1.7.1 producing the following traceback:

```

Error
Traceback (most recent call last):
  File "lib\site-packages\sqlalchemy\engine\base.py", line 1276, in _execute_context
    self.dialect.do_execute(
  File "lib\site-packages\sqlalchemy\engine\default.py", line 608, in do_execute
    cursor.execute(statement, parameters)
sqlite3.OperationalError: no such column: object_class_id

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "data\tests\test_migration.py", line 102, in test_upgrade_content
    db_map = DiffDatabaseMapping(db_url, upgrade=True)
  File "data\spinedb_api\diff_db_mapping.py", line 54, in __init__
    super().__init__(*args, **kwargs)
  File "spinedb_api\db_mapping_add_mixin.py", line 32, in __init__
    super().__init__(*args, **kwargs)
  File "data\spinedb_api\diff_db_mapping_base.py", line 43, in __init__
    super().__init__(*args, **kwargs)
  File "data\spinedb_api\db_mapping_base.py", line 80, in __init__
    self.engine = self.create_engine(self.sa_url, upgrade=upgrade, create=create)
  File "data\spinedb_api\db_mapping_base.py", line 309, in create_engine
    environment_context.run_migrations()
  File "lib\site-packages\alembic\runtime\environment.py", line 851, in run_migrations
    self.get_context().run_migrations(**kw)
  File "lib\site-packages\alembic\runtime\migration.py", line 612, in run_migrations
    step.migration_fn(**kw)
  File "data\spinedb_api\alembic\versions\bba1e2ef5153_move_to_entity_based_design.py", line 466, in upgrade
    alter_tables_after_update(meta)
  File "data\spinedb_api\alembic\versions\bba1e2ef5153_move_to_entity_based_design.py", line 440, in alter_tables_after_update
    batch_op.drop_column(dummy_relationship_class)
  File "lib\contextlib.py", line 120, in __exit__
    next(self.gen)
  File "lib\site-packages\alembic\operations\base.py", line 369, in batch_alter_table
    impl.flush()
  File "lib\site-packages\alembic\operations\batch.py", line 140, in flush
    batch_impl._create(self.impl)
  File "lib\site-packages\alembic\operations\batch.py", line 411, in _create
    op_impl.create_table(self.new_table)
  File "lib\site-packages\alembic\ddl\impl.py", line 356, in create_table
    self._exec(schema.CreateTable(table))
  File "lib\site-packages\alembic\ddl\impl.py", line 197, in _exec
    return conn.execute(construct, multiparams)
  File lib\site-packages\sqlalchemy\engine\base.py", line 1011, in execute
    return meth(self, multiparams, params)
  File "lib\site-packages\sqlalchemy\sql\ddl.py", line 72, in _execute_on_connection
    return connection._execute_ddl(self, multiparams, params)
  File "lib\site-packages\sqlalchemy\engine\base.py", line 1068, in _execute_ddl
    ret = self._execute_context(
  File "lib\site-packages\sqlalchemy\engine\base.py", line 1316, in _execute_context
    self._handle_dbapi_exception(
  File "lib\site-packages\sqlalchemy\engine\base.py", line 1510, in _handle_dbapi_exception
    util.raise_(
  File "lib\site-packages\sqlalchemy\util\compat.py", line 182, in raise_
    raise exception
  File "lib\site-packages\sqlalchemy\engine\base.py", line 1276, in _execute_context
    self.dialect.do_execute(
  File "lib\site-packages\sqlalchemy\engine\default.py", line 608, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) no such column: object_class_id
[SQL: 
CREATE TABLE _alembic_tmp_parameter_definition (
	id INTEGER NOT NULL, 
	name VARCHAR(155) NOT NULL, 
	description VARCHAR(155) DEFAULT (NULL), 
	data_type VARCHAR(155) DEFAULT 'NUMERIC', 
	default_value VARCHAR(155) DEFAULT '0', 
	commit_id INTEGER, 
	parameter_value_list_id INTEGER, 
	entity_class_id INTEGER NOT NULL, 
	CONSTRAINT pk_parameter PRIMARY KEY (id), 
	CONSTRAINT fk_parameter_commit_id_commit FOREIGN KEY(commit_id) REFERENCES "commit" (id), 
	CONSTRAINT uq_parameter_definition_nameentity_class_id UNIQUE (name, entity_class_id), 
	CONSTRAINT ck_parameter_obj_or_rel_class_id_is_not_null CHECK (`relationship_class_id` IS NOT NULL OR `object_class_id` IS NOT NULL), 
	CONSTRAINT fk_parameter_definition_entity_class_id_entity_class FOREIGN KEY(entity_class_id) REFERENCES entity_class (id) ON DELETE CASCADE ON UPDATE CASCADE
)

]
(Background on this error at: http://sqlalche.me/e/13/e3q8)
```

**Comment by PekkaSavolainen** [comment]
Which version of alembic should we fix for the upcoming release?

**Comment by soininen** [comment]
> Which version of alembic should we fix for the upcoming release?

I just fixed the migration scripts so maybe alembic 1.7.1?

**Comment by manuelma** [comment]
I'm not sure about 'fixing' a migration script. I would rather fix the version of alembic that doesn't need the script fix. It's always too late to fix a migration script that might already be applied to a lot of databases. Those databases will never see that fix. Conversely, some databases might see the fix. So in the end, we would have to subsets of databases that we need to account for in the future. What do you think?

**Comment by soininen** [comment]
I'd agree, if the 'fixed' migration produced different results. In this case, there is no difference as both alembic 1.6 and 1.7 drop the same constraints. So I say we keep the fix and require alembic>1.7.

**Comment by manuelma** [comment]
Ah, thanks for that. I wasn't 100% sure that that was the case although it looked like it. So we're good, thanks @soininen !
